src/layers/mlp_chatglm2.h

```diff
@@ -37,37 +37,68 @@ class ChatGLM2MLP : public LlamaMLP<WeiT> {
 
         auto range = SplitUtil::getTaskRange(intermediateSize, ctx->numSplit, ctx->splitIdx);
         int colSplit = range.second - range.first;
-        float *gateW = (float *)malloc(hiddenSize * colSplit * sizeof(float));
-        float *upW = (float *)malloc(hiddenSize * colSplit * sizeof(float));
-        if (trans) {
-            int blockSize = colSplit * hiddenSize;
-            memcpy(gateW, gate_upW + ctx->splitIdx * blockSize, blockSize * sizeof(float));
-            memcpy(upW, gate_upW + intermediateSize * hiddenSize + ctx->splitIdx * blockSize,
-                    blockSize * sizeof(float));
-        } else {
-            const float *weightPTR = gate_upW;
-            for (int i = 0; i < hiddenSize; i++) {
-                memcpy(gateW + i * colSplit, weightPTR + ctx->splitIdx * colSplit, colSplit * sizeof(float));
-                weightPTR += intermediateSize;
-                memcpy(upW + i * colSplit, weightPTR + ctx->splitIdx * colSplit, colSplit * sizeof(float));
-                weightPTR += intermediateSize;
+
+        int enable = (getenv("ENABLE_CAT_MLP") ? atoi(getenv("ENABLE_CAT_MLP")) : 1);
```

Comment:
why we provide the option to disable this?

Reply:
Leave the option temporarily for future performance tuning. Maybe we can delete it after a while.

Reply:
maybe a better method is to use a static variable, thus do not need to call 'getenv' every time.
(and if it is used multiple times in the file, we may declare a global static variable)

Reply:
I added global vars in mlp_llama.cpp. Please review~